relay-server/src/utils/sleep_handle.rs

```diff
@@ -35,9 +36,15 @@ impl Future for SleepHandle {
     type Output = ();
 
     fn poll(mut self: Pin<&mut Self>, cx: &mut std::task::Context<'_>) -> Poll<Self::Output> {
-        match &mut self.0 {
+        let poll = match &mut self.0 {
             Some(sleep) => Pin::new(sleep).poll(cx),
             None => Poll::Pending,
+        };
+
+        if poll.is_ready() {
+            self.reset();
         }
```

Comment:
At some point during the implementation, I forgot to reset the sleep handle after polling it successfully. That sent the service into an endless loop. That is a dangerous API. Usually, the expectation can be that once the sleep has been polled successfully, it is handled and can be reset.

Reply:
please update the doc comment to explain that the reset happens automatically